xous-rs/src/definitions/messages/envelope.rs

```diff
@@ -124,21 +124,19 @@ impl Envelope {
                 ));
             }
 
-            _ => {
+            Message::Borrow(_) | Message::MutableBorrow(_) | Message::Scalar(_) => {
                 let result = crate::send_message(connection, body);
                 let new_self = ManuallyDrop::into_inner(manual_self);
-
-                if let Ok(crate::Result::Ok) = result {
+                match result {
                     // `new_self` will have its Drop() called
-                    return Ok(());
-                }
+                    Ok(crate::Result::Ok) | Ok(crate::Result::MemoryReturned(_, _)) => Ok(()),
```

Comment:
ah. right..."all you have to do" is drop a memory message for it to return. Looks right to me.